Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyComboBox.cxx

```diff
@@ -299,21 +308,60 @@ void qMRMLSubjectHierarchyComboBox::showPopup()
   QPoint below = mapToGlobal(listRect.bottomLeft());
   QPoint above = mapToGlobal(listRect.topLeft());
 
-  // Custom size
+  // Custom Height
+  int popupHeight = 0;
   int displayedItemCount = d->TreeView->displayedItemCount();
-  const int numberOfRows = qMin(displayedItemCount, d->MaximumNumberOfShownItems);
-  QSize itemSize = QSize(
-    d->TreeView->sizeHintForColumn(d->TreeView->model()->nameColumn()), d->TreeView->sizeHintForRow(0) );
-  listRect.setHeight( numberOfRows * itemSize.height() );
+  if (displayedItemCount == 0)
+    {
+    // If not items, find what message to show instead
+    vtkMRMLSubjectHierarchyNode* shNode = d->TreeView->subjectHierarchyNode();
+    vtkIdType rootItem = d->TreeView->rootItem();
+    std::vector<vtkIdType> childItemIDs;
+    shNode->GetItemChildren(rootItem, childItemIDs, false);
+    if (childItemIDs.empty())
+      {
+      if (rootItem!= shNode->GetSceneItemID())
+        {
+        d->NoItemLabel->setText("No items in branch");
```

Comment:
@cpinter the only thing I wonder is, maybe we could display the name of the root item instead of `branch` here?